common/src/main/java/org/jboss/pnc/common/monitor/CancellableCompletableFuture.java

```diff
@@ -15,14 +15,26 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-
 package org.jboss.pnc.common.monitor;
 
+import java.util.concurrent.CompletableFuture;
+
 /**
+ * CompletableFuture with task cancellation support.
+ *
  * @author <a href="mailto:matejonnet@gmail.com">Matej Lazar</a>
  */
-public class MonitorException extends RuntimeException {
-    public MonitorException(String message) {
-        super(message);
+public class CancellableCompletableFuture<T> extends CompletableFuture<T> {
+
+    private final Runnable onCancel;
+
+    public CancellableCompletableFuture(Runnable onCancel) {
```

Comment:
Could you document  this method that says that the runnable is invoked only when 'cancel' is called? It's pretty obvious now with the argument name, but on initial read that confused me a little bit.